src/mlpack/methods/logistic_regression/logistic_regression_main.cpp

```diff
@@ -328,6 +347,19 @@ static void mlpackMain()
       CLI::GetParam<arma::Row<size_t>>("output") = std::move(predictions);
```

Comment:
This `CLI::GetParam<...>` call is the only line that needs to be in the `if` statement.  Can you condense this (and the following bit of code) a little accordingly please?

Reply:
I mean this...

```
if (CLI::HasParam("output") || CLI::HasParam("predictions"))
{
  Log::Info << "Predicting classes of points in '"
      << CLI::GetPrintableParam<arma::mat>("test") << "'." << endl;
  model->Classify(testSet, predictions, decisionBoundary);

  // 'output' and 'predictions' are the same, so output to both.
  CLI::GetParam<arma::Row<size_t>>("output") = predictions;
  CLI::GetParam<arma::Row<size_t>>("predictions") = std::move(predictions);
}
```

What do you think, did I overlook anything?

Reply:
I am not sure on how this code would behave in case when param `output` is not defined and we try to store values in `output` as well as parameters. It might give error: param `output` not defined.
I'll still try this code. Thanks for your reply :)

Reply:
So actually the code should not have an error, since we can call `CLI::GetParam<...>("output")` even if `CLI::HasParam("output")` is false---the only requirement is that there is a `PARAM_*("output", ...)` somewhere.

But, now that I think about it, we shouldn't set those outputs if the user didn't ask for them---`CLI::HasParam("output")` will only be true if the user has asked for the output.  So we can make a simple change---

```
if (CLI::HasParam("output"))
  CLI::GetParam<arma::Row<size_t>>("output") = predictions;
if (CLI::HasParam("predictions"))
  CLI::GetParam<arma::Row<size_t>>("predictions") = std::move(predictions);
```

Reply:
That looks good, will update it :)